tests/integrations/download/test_Download__file.py

```diff
@@ -86,6 +89,11 @@ def test_new_download_oneshot(mock_tools, url, content_disposition):
     # The filename is derived from the URL or header
     assert filename == mock_tools.base_path / "downloads" / "something.zip"
 
+    # Temporary file was upgraded to intended destination
+    mock_tools.shutil.move.assert_called_with(mock.ANY, filename)
```

Comment:
While this test passes, it's not especially resilient - `mock.ANY` really does allow *any* content. This initial "mock.ANY" call is definitely required to verify that *a* filename was used, but it would be desirable to add checks that the same temporary filename was used for all the subsequent calls.

Reply:
That's fair; I was initially struggling with how to introspect information about the temporary file and I went too far towards "assume it's working if everything else is working." I think my changes now verify the logic is working as designed.